staging/src/k8s.io/cli-runtime/pkg/printers/warningprinter.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"fmt"
+	"io"
+	"sync"
+)
+
+const (
+	yellowColor = "\u001b[33;1m"
+	resetColor  = "\u001b[0m"
+)
+
+type WarningPrinter struct {
+	// out is the writer to output warnings to
+	out io.Writer
+	// opts contains options controlling warning output
+	opts WarningPrinterOptions
+	// writtenLock guards written and writtenCount
+	writtenLock  sync.Mutex
+	writtenCount int
+	written      map[string]struct{}
+}
+
+// WarningPrinterOptions controls the behavior of a WarningPrinter constructed using NewWarningPrinter()
+type WarningPrinterOptions struct {
+	// Deduplicate indicates a given warning message should only be written once.
+	// Setting this to true in a long-running process handling many warnings can result in increased memory use.
+	Deduplicate bool
+	// Color indicates that warning output can include ANSI color codes
+	Color bool
+}
+
+// NewWarningPrinter returns an implementation of warningPrinter that outputs warnings to the specified writer.
+func NewWarningPrinter(out io.Writer, opts WarningPrinterOptions) *WarningPrinter {
+	h := &WarningPrinter{out: out, opts: opts}
+	if opts.Deduplicate {
+		h.written = map[string]struct{}{}
+	}
+	return h
+}
+
+// Print prints warnings to the configured writer.
+func (w *WarningPrinter) Print(message string) {
+
+	w.writtenLock.Lock()
+	defer w.writtenLock.Unlock()
+
+	if w.opts.Deduplicate {
+		if _, alreadyWritten := w.written[message]; alreadyWritten {
+			return
+		}
+		w.written[message] = struct{}{}
+	}
+	w.writtenCount++
```

Comment:
Did you intend for this to be used anywhere?

Reply:
thanks for the comments. I may not get your point. Did you mean the deduplication and count is not needed, as it would not be used anywhere?

Reply:
It does not look like `writtenCount` is used anywhere other than the increment here. Not sure if you had plans to use this int.

Reply:
Sorry for the confusion, I have no plan for it. 

Removed.